### PR TITLE
doc and cleanup pass

### PR DESCRIPTION
**Pull request by dhalbert**
I made a pass over all the files to do a more thorough code review, made some changes, and updated some doc. In general:

-- Use of `kwargs` was mostly dropped in favor of explicit arg names.
-- In a few places, the fact that an object was from `_bleio` was made more explicit.
-- Some pylint overrides were removed by renaming variables, etc.
-- A couple of pylint warnings were dealt with.
-- `_bleio.Attribute` and `_bleio.Characteristic` CONSTANT values were duplicated in the library, so that user code and library code does not need to refer to these directly.
-- 8, 16, and 32 bit signed and unsigned characteristic classes were created.
- Fixed an apparent bug in `time_struct` stuff, caught by @jerryneedell.

Tested `ble_uart_echo_test.py` and`ble_detailed_scan.py`.
Tested `ble_demo_central.py` and `ble_demo_periph.py` working together on a pair of CPB's.

There is one minor corresponding fix to circuitpython which I'll submit a PR for shortly.

### CODE REVIEW

File: adafruit_ble/__init__.py [hunk 1]
```diff
@@ -53,46 +54,60 @@
...
-       It acts as a map from a Service type to a Service instance for the connection.
     """
-    def __init__(self, connection):
-        self._connection = connection
-        self._discovered_services = {}
-        """These are the bare remote services from _bleio."""
+    Represents a connection to a peer BLE device.
+    It acts as a map from a `Service` type to a `Service` instance for the connection.
 
+    :param bleio_connection _bleio.connection
```
Comment (tannewt) [inline adafruit_ble/__init__.py]: This doc looks incomplete.

File: adafruit_ble/__init__.py [hunk 2]
```diff
@@ -53,46 +54,60 @@
...
+    """
+    def __init__(self, bleio_connection):
+        self._bleio_connection = bleio_connection
+        # _bleio.Service objects representing services found during discoery.
```
Comment (tannewt) [inline adafruit_ble/__init__.py]:
```suggestion
        # _bleio.Service objects representing services found during discovery.
```

File: adafruit_ble/__init__.py [hunk 3]
```diff
@@ -167,7 +213,14 @@ def stop_scan(self):
...
 
     def connect(self, advertisement, *, timeout=4):
-        """Initiates a `BLEConnection` to the peer that advertised the given advertisement."""
+        """
+        Initiates a `BLEConnection` to the peer that advertised the given advertisement.
+
+        :param advertisement Advertisement: An `Advertisement` or a subclass of `Advertisement`
+        :param timeout float: how long to wait for a connection
+        :returns the connection to the peer
+        :rtype BLEConnection
```
Comment (tannewt) [inline adafruit_ble/__init__.py]: This work? I thought it should ble `:returns:`.

**Comment by jerryneedell** [comment]
Don’t these references to “core” have to be fixed? https://github.com/adafruit/Adafruit_CircuitPython_BLE/blob/dcf76c7bffd809c88d7a631adba858dccdcc2289/adafruit_ble/services/standard/standard.py#L32

File: adafruit_ble/__init__.py (hunk 1 above)
Comment (dhalbert) [reply adafruit_ble/__init__.py]: Most of the doc is in the method docs.

File: adafruit_ble/__init__.py (hunk 3 above)
Comment (dhalbert) [reply adafruit_ble/__init__.py]: You're right. I find Sphinx to be very non-obvious.
https://github.com/adafruit/Adafruit_nRF52_Arduino/blob/develop/libraries/Bluefruit52Lib/examples/Peripheral/cplay_ble/services.txt

Also the `rtype` must be specified separately from `return` or `returns`, unlike `param`.

Comment (dhalbert) [reply adafruit_ble/__init__.py]: Also a `:return:` for a constructor is kinda obvious. I removed that one.

**Comment by dhalbert** [comment]
@jerryneedell Yes, thanks, that code wasn't used, so it didn't get vetted.

File: adafruit_ble/__init__.py (hunk 1 above)
Comment (dhalbert) [reply adafruit_ble/__init__.py]: Let me know if you want more and what it would be.

**Comment by jerryneedell** [comment]
hmm -- tried simpletst ```
Press any key to enter the REPL. Use CTRL-D to reload.
Adafruit CircuitPython 5.0.0-alpha.5-80-g166518fc9 on 2019-11-13; Adafruit Feather nRF52840 Express with nRF52840
>>> import ble_simpletest
scanning
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "ble_simpletest.py", line 10, in <module>
  File "adafruit_ble/__init__.py", line 198, in start_scan
TypeError: can't convert NoneType to float
>>> 
```

**Comment by dhalbert** [comment]
@jerryneedell This is fixed by https://github.com/adafruit/circuitpython/pull/2287, so you need a newer CPy.

**Comment by jerryneedell** [comment]
Ah -- so it is!! Sorry - I missed that commit to CP. works fine

File: adafruit_ble/services/standard/standard.py [hunk 4]
```diff
@@ -29,10 +29,10 @@
 
 import time
 
-from ..core import Service
-from ...core.uuid import StandardUUID
+from .. import Service
+from ...uuid import StandardUUID
 from ..characteristics.string import StringCharacteristic
-from ..characteristics.core import StructCharacteristic
+from ..characteristics import StructCharacteristic
```
Comment (jerryneedell) [inline adafruit_ble/services/standard/standard.py]: FYI -- each of these needs one more "."   ```...characteristics```

Comment (jerryneedell) [reply adafruit_ble/services/standard/standard.py]: Still returns all 0's ,but with that change, it does not throw any errors ;-)

Comment (dhalbert) [reply adafruit_ble/services/standard/standard.py]: Oops, I will push another change, but what returns all 0's?

Comment (jerryneedell) [reply adafruit_ble/services/standard/standard.py]: I've been running this -- modified from your old version  
```
"""CTS demo
"""

import board
import displayio
import terminalio
import time

import adafruit_displayio_ssd1306
from adafruit_display_text import label
from adafruit_ble import BLERadio
from adafruit_ble.advertising.standard import ProvideServicesAdvertisement
from adafruit_ble.services.standard.standard import CurrentTimeService

ble = BLERadio()
cts = CurrentTimeService()
advertisement = ProvideServicesAdvertisement(cts)



displayio.release_displays()

i2c = board.I2C()
display_bus = displayio.I2CDisplay(i2c, device_address=0x3c)
display = adafruit_displayio_ssd1306.SSD1306(display_bus, width=128, height=32)

FONT = terminalio.FONT
WHITE = 0xffffff
CONNECTING = "Connecting..."
text = CONNECTING

text_area = label.Label(FONT, max_glyphs=50, text=text, color=WHITE)

text_area.x = 0
text_area.y = 10


display.show(text_area)

while True:
    text_area.text = CONNECTING
    ble.start_advertising(advertisement)
    while not ble.connected:
        pass

    while ble.connected:
        print(cts.struct_time)
        year, month, day, hour, minute, second, _, _, _ = cts.current_time
        text_area.text = '{}-{}-{} {:02}:{:02}:{:02}'.format(year, month, day, hour, minute, second)
        time.sleep(5)

```

I don't get the pairing option on Settings on my iPhone, but if I connect via the BLE Conect App it connects and but the time struct is all 0's 
```
Press any key to enter the REPL. Use CTRL-D to reload.
Adafruit CircuitPython 5.0.0-alpha.5-84-g4c55431bb on 2019-11-14; Adafruit Feather nRF52840 Express with nRF52840
>>> import cts
struct_time(tm_year=0, tm_mon=0, tm_mday=0, tm_hour=0, tm_min=0, tm_sec=0, tm_wday=-1, tm_yday=-1, tm_isdst=-1)
```

Comment (jerryneedell) [reply adafruit_ble/services/standard/standard.py]: This does not have to be resolved for this PR -- no need to hold things up.

**Comment by jerryneedell** [comment]
Tested central/periph demo on CPBs  and uart_echo_test,ble_simpletest and ble_detailed_scan on feather_nrf52840.  Latest fixes to standard.py fix the errors I was seeing - still experimenting with CTS but not working yet.

**Pull request merged**